pkg/types/validation/installconfig.go

```diff
@@ -555,3 +582,26 @@ func validateCloudCredentialsMode(mode types.CredentialsMode, fldPath *field.Pat
 	}
 	return allErrs
 }
+
+// validateURI checks if the given url is of the right format.
+func validateURI(uri string, fldPath *field.Path) (string, field.ErrorList) {
+	allErrs := field.ErrorList{}
+	parsed, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return "", append(allErrs, field.Invalid(fldPath, uri, err.Error()))
+	}
+	if !parsed.IsAbs() {
+		return "", append(allErrs, field.Invalid(fldPath, uri, fmt.Sprintf("failed validating URI, no scheme for URI")))
```

Comment:
I would add this error to `allErrs` and continue rather than returning here. Ultimately, I think that `validateURI` should take a list of acceptable schemes and validate the scheme rather than returning the scheme. The `validateURI` function should return all errors rather than just the first error encountered.